internal/integration/knot/knot_protocol.go

```diff
@@ -370,13 +374,18 @@ func dataControl(deviceChan chan entities.Device, p *protocol) {
 				if err != nil {
 					log.WithFields(log.Fields{"knot": entities.KnotError}).Error(err)
 				} else {
-					err = p.network.publisher.PublishDeviceData(p.userToken, &device, device.Data)
-					if err != nil {
-						log.WithFields(log.Fields{"knot": entities.KnotError}).Error(err)
-					} else {
-						device.Data = nil
-						p.updateDevice(device)
-						log.WithFields(log.Fields{"knot": entities.KnotAuth}).Info("send the new data comes from the device")
+					datas := device.Data
+					device.Data = nil
+					for _, data := range datas {
```

Comment:
Here the sensor slice is reset so that each sensor is sent at a time? If so, what do you think about moving this code snippet to a function that takes "device" as an argument? Also, consider adding comments that explain the logic of this snippet.